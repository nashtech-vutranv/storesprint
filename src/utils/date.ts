import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
