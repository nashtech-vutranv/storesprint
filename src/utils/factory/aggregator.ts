import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
