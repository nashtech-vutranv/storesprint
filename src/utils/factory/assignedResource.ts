import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
