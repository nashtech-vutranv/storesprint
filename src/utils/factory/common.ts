import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
