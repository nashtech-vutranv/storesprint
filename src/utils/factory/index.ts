export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
