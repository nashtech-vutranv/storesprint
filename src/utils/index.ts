export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
