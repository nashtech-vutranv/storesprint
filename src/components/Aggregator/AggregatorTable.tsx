import React, {useState, useEffect, useContext} from 'react'
import {useTranslation} from 'react-i18next'
import {Column} from 'primereact/column'
import {Button as ButtonPrime} from 'primereact/button'
import {DataTable} from 'primereact/datatable'
import {InputText} from 'primereact/inputtext'
import {Button, Row, Col} from 'react-bootstrap'
import {GlobalContext} from '../../store/GlobalContext'
import {usePagination, useCommonAccesibility, useHandleError} from '../../hooks'
import AggregatorService from '../../services/AggregatorService'
import {capitalizeFirstLetter} from '../../helpers/characters'
import {IAggregator} from '../../interface/aggregator'
import {TemplatePaginator} from '../Paginator'
import FieldTextDataTable from '../FieldTextDataTable/FieldTextDataTable'
import {RowTableActionType} from '../../store/actions'
import {defaultAggregatorsTablePaginationPerPage} from '../../constants/pagination'

export default function AggregatorTable() {
  const {t} = useTranslation()
  const {
    state: {axiosClient},
    dispatch: {rowTable: rowTableDispatch},
  } = useContext(GlobalContext)

  const {handleErrorResponse} = useHandleError()

  const {
    dataApi: {keyword, pagination, defaultPagination},
    dataTable: {
      totalRecords,
      setTotalRecords,
      onPage,
      onSort,
      onSearch,
      onKeywordChange,
      setPagination,
      setKeyword,
      onBlurInputSearch,
    },
  } = usePagination(defaultAggregatorsTablePaginationPerPage)

  const [aggregators, setAggregators] = useState<IAggregator[]>([])

  useEffect(() => {
    if (pagination) {
      new AggregatorService(axiosClient)
      .fetchAllAggregators(pagination, {
        search: keyword,
      })
      .then((response: any) => {
        const mappingData = response.data
        const aggregatorsData = mappingData.content.map((item: any) => ({
          ...item,
          status: capitalizeFirstLetter(item.status),
        }))
        setAggregators(aggregatorsData)
        setTotalRecords(mappingData.totalElements)
      })
      .catch((err: any) => {
        handleErrorResponse(err)
      })
    }
  }, [pagination])

  useCommonAccesibility()

  const refreshInitialState = () => {
    setKeyword('')
    pagination && setPagination({
      ...defaultPagination,
      sortField: pagination?.sortField,
      sortOrder: pagination?.sortOrder,
    })
    rowTableDispatch({
      type: RowTableActionType.GET_CURRENT_TABLE_ROW_QUANTITIES,
      payload: 10,
    })
  }

  const renderHeader = () => {
    return (
      <Row>
        <div className='d-flex justify-content-start align-items-center'>
          <Col xs={8} sm={8} md={8} lg={8} xl={8} xxl={8} className='btn-mr-1'>
            <InputText
              value={keyword}
              onChange={onKeywordChange}
              placeholder={t('common_input_search_placeHolder')}
              className='w-100'
              onBlur={onBlurInputSearch}
              onKeyDown={(event: any) => event.key === 'Enter' && onSearch()}
            />
          </Col>
          <Button onClick={onSearch} className='btn-h-95'>
            {t('common_button_search_label')}
          </Button>
        </div>
      </Row>
    )
  }

  const templateLamda = TemplatePaginator('sample_lamda')

  const paginatorLeft = (
    <ButtonPrime
      aria-label='refresh-button'
      type='button'
      icon='pi pi-refresh'
      className='p-button-text refresh-button'
      onClick={refreshInitialState}
      tooltip={t('common_button_refresh_table')}
    />
  )

  return (
    <>
      <DataTable
        sortField={pagination?.sortField}
        sortOrder={pagination?.sortOrder}
        totalRecords={totalRecords}
        dataKey='id'
        value={aggregators}
        paginator
        paginatorTemplate={templateLamda}
        first={pagination?.first}
        rows={pagination?.rows}
        responsiveLayout='scroll'
        selectionMode='single'
        header={renderHeader}
        className='data-table-mh'
        paginatorClassName='table-paginator'
        emptyMessage={t('aggregator_table_empty_message')}
        paginatorLeft={paginatorLeft}
        lazy
        onSort={onSort}
        onPage={onPage}
      >
        <Column
          field='name'
          header={t('aggregator_column_header_name')}
          sortable
          body={(agg: IAggregator) => (
            <FieldTextDataTable value={agg.name} placement='bottom' />
          )}
        ></Column>
        <Column
          field='status'
          header={t('aggregator_column_header_status')}
          sortable
        ></Column>
      </DataTable>
    </>
  )
}
