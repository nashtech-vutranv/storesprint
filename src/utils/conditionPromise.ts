import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}