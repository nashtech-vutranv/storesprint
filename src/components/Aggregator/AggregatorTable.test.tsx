import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import {mockAggregators} from '../../utils/factory'
import AggregatorTable from './AggregatorTable'

const tree = render(
  <MemoryRouter>
    <AggregatorTable />
  </MemoryRouter>
)

jest.mock('../../services/AggregatorService', () => {
  const instance = jest.fn().mockReturnValue({
    fetchAllAggregators: jest.fn().mockImplementation(() => {
      return Promise.resolve(mockAggregators())
    }),
  })

  return instance
})

describe('Aggregator Table', () => {
  it('should match snapshot', async () => {
      waitFor(() => {
        expect(tree).toMatchSnapshot()
      })
    })
})
