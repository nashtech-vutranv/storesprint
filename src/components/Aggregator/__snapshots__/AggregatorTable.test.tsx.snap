// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aggregator Table should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="row"
          >
            <div
              class="d-flex justify-content-start align-items-center"
            >
              <div
                class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
              >
                <input
                  class="p-inputtext p-component w-100"
                  placeholder="common_input_search_placeHolder"
                  value=""
                />
              </div>
              <button
                class="btn-h-95 btn btn-primary"
                type="button"
              >
                common_button_search_label
              </button>
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="ascending"
                  class="p-sortable-column p-highlight"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      aggregator_column_header_name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      aggregator_column_header_status
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-selectable-row"
                role="row"
                tabindex="0"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Account
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  Z,uskb+j\`3
                </td>
              </tr>
              <tr
                class="p-selectable-row p-row-odd"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Boliviano
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  Qp^|u90m&lt;g
                </td>
              </tr>
              <tr
                class="p-selectable-row"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    bypassing
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  T/-&ynp.n"
                </td>
              </tr>
              <tr
                class="p-selectable-row p-row-odd"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    bandwidth
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  Qo0[9z0h1h
                </td>
              </tr>
              <tr
                class="p-selectable-row"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    CSS
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  Kdseg2:6"v
                </td>
              </tr>
              <tr
                class="p-selectable-row p-row-odd"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Principal
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  &lt;eh12mp6x2
                </td>
              </tr>
              <tr
                class="p-selectable-row"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Kwacha
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  !jwi!={9@^
                </td>
              </tr>
              <tr
                class="p-selectable-row p-row-odd"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Oriental
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  Q&|lx&lt;t3lf
                </td>
              </tr>
              <tr
                class="p-selectable-row"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    Chicken
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  /#1a&gt;h%c=7
                </td>
              </tr>
              <tr
                class="p-selectable-row p-row-odd"
                role="row"
                tabindex="-1"
              >
                <td
                  class=""
                  role="cell"
                >
                  <div
                    class="linesEllipsis"
                    style="text-align: left;"
                  >
                    navigating
                  </div>
                </td>
                <td
                  class=""
                  role="cell"
                >
                  H?2c%tvyk!
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text refresh-button p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                Â 
              </span>
            </button>
          </div>
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_first
            </span>
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_previous
            </span>
          </button>
          <span
            class="p-paginator-pages"
          >
            <button
              class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-paginator-page-end p-highlight"
              type="button"
            >
              1
            </button>
          </span>
          <button
            class="p-paginator-next p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_next
            </span>
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_last
            </span>
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper dropdown-table-input  p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                aria-label="textbox"
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="dropdown-trigger"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            style="user-select: none; text-align: center; margin-left: auto;"
          >
            organization_paginator_show_report
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-datatable p-component p-datatable-selectable p-datatable-responsive-scroll data-table-mh"
      data-scrollselectors=".p-datatable-wrapper"
      pr_id_1=""
    >
      <div
        class="p-datatable-header"
      >
        <div
          class="row"
        >
          <div
            class="d-flex justify-content-start align-items-center"
          >
            <div
              class="btn-mr-1 col-xxl-8 col-xl-8 col-lg-8 col-md-8 col-sm-8 col-8"
            >
              <input
                class="p-inputtext p-component w-100"
                placeholder="common_input_search_placeHolder"
                value=""
              />
            </div>
            <button
              class="btn-h-95 btn btn-primary"
              type="button"
            >
              common_button_search_label
            </button>
          </div>
        </div>
      </div>
      <div
        class="p-datatable-wrapper"
      >
        <table
          class="p-datatable-table"
          role="table"
        >
          <thead
            class="p-datatable-thead"
          >
            <tr
              role="row"
            >
              <th
                aria-sort="ascending"
                class="p-sortable-column p-highlight"
                role="columnheader"
                tabindex="0"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    aggregator_column_header_name
                  </span>
                  <span
                    class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                  />
                </div>
              </th>
              <th
                aria-sort="none"
                class="p-sortable-column"
                role="columnheader"
                tabindex="0"
              >
                <div
                  class="p-column-header-content"
                >
                  <span
                    class="p-column-title"
                  >
                    aggregator_column_header_status
                  </span>
                  <span
                    class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="p-datatable-tbody"
          >
            <tr
              class="p-selectable-row"
              role="row"
              tabindex="0"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Account
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                Z,uskb+j\`3
              </td>
            </tr>
            <tr
              class="p-selectable-row p-row-odd"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Boliviano
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                Qp^|u90m&lt;g
              </td>
            </tr>
            <tr
              class="p-selectable-row"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  bypassing
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                T/-&ynp.n"
              </td>
            </tr>
            <tr
              class="p-selectable-row p-row-odd"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  bandwidth
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                Qo0[9z0h1h
              </td>
            </tr>
            <tr
              class="p-selectable-row"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  CSS
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                Kdseg2:6"v
              </td>
            </tr>
            <tr
              class="p-selectable-row p-row-odd"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Principal
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                &lt;eh12mp6x2
              </td>
            </tr>
            <tr
              class="p-selectable-row"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Kwacha
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                !jwi!={9@^
              </td>
            </tr>
            <tr
              class="p-selectable-row p-row-odd"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Oriental
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                Q&|lx&lt;t3lf
              </td>
            </tr>
            <tr
              class="p-selectable-row"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  Chicken
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                /#1a&gt;h%c=7
              </td>
            </tr>
            <tr
              class="p-selectable-row p-row-odd"
              role="row"
              tabindex="-1"
            >
              <td
                class=""
                role="cell"
              >
                <div
                  class="linesEllipsis"
                  style="text-align: left;"
                >
                  navigating
                </div>
              </td>
              <td
                class=""
                role="cell"
              >
                H?2c%tvyk!
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="p-paginator p-component p-paginator-bottom table-paginator"
      >
        <div
          class="p-paginator-left-content"
        >
          <button
            aria-label="refresh-button"
            class="p-button p-component p-button-text refresh-button p-button-icon-only"
            type="button"
          >
            <span
              class="p-button-icon p-c pi pi-refresh"
            />
            <span
              class="p-button-label p-c"
            >
              Â 
            </span>
          </button>
        </div>
        <button
          class="p-paginator-first p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-2"
          >
            organization_paginator_first
          </span>
        </button>
        <button
          class="p-paginator-prev p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-3"
          >
            organization_paginator_previous
          </span>
        </button>
        <span
          class="p-paginator-pages"
        >
          <button
            class="p-paginator-page p-paginator-element p-link p-paginator-page-start p-paginator-page-end p-highlight"
            type="button"
          >
            1
          </button>
        </span>
        <button
          class="p-paginator-next p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-3"
          >
            organization_paginator_next
          </span>
        </button>
        <button
          class="p-paginator-last p-paginator-element p-link p-disabled"
          disabled=""
          type="button"
        >
          <span
            class="p-2"
          >
            organization_paginator_last
          </span>
        </button>
        <div
          class="p-dropdown p-component p-inputwrapper dropdown-table-input  p-inputwrapper-filled"
        >
          <div
            class="p-hidden-accessible"
          >
            <input
              aria-haspopup="listbox"
              aria-label="textbox"
              readonly=""
              type="text"
            />
          </div>
          <div
            class="p-hidden-accessible p-dropdown-hidden-select"
          >
            <select
              aria-hidden="true"
              tabindex="-1"
            >
              <option
                value=""
              />
              <option
                value="10"
              >
                10
              </option>
            </select>
          </div>
          <span
            class="p-dropdown-label p-inputtext"
          >
            10
          </span>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="dropdown-trigger"
            class="p-dropdown-trigger"
            role="button"
          >
            <span
              class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
            />
          </div>
        </div>
        <span
          style="user-select: none; text-align: center; margin-left: auto;"
        >
          organization_paginator_show_report
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
